x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsTaskState.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.dataframe;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.persistent.PersistentTaskState;
+import org.elasticsearch.persistent.PersistentTasksCustomMetaData;
+import org.elasticsearch.xpack.core.ml.MlTasks;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class DataFrameAnalyticsTaskState implements PersistentTaskState {
+
+    private static ParseField STATE = new ParseField("state");
+    private static ParseField ALLOCATION_ID = new ParseField("allocation_id");
+
+    private final DataFrameAnalyticsState state;
+    private final long allocationId;
+
+    private static final ConstructingObjectParser<DataFrameAnalyticsTaskState, Void> PARSER =
+            new ConstructingObjectParser<>(MlTasks.DATA_FRAME_ANALYTICS_TASK_NAME,
```

Comment:
I would suggest making this parser ignore unknown fields.  We're making that change to the `JobTaskState` in #37982 so that we can add a failure reason in 7.0.  It's possible we could want to add more fields to this task state in the future too.

Reply:
Will do!